pkgs/applications/science/logic/coq/default.nix

```diff
@@ -56,8 +56,8 @@ let
     args.version;
   version = fetched.version;
   coq-version = args.coq-version or (if version != "dev" then versions.majorMinor version else "dev");
-  versionAtLeast = v: (coq-version == "dev") || (lib.versionAtLeast coq-version v);
-  ideFlags = optionalString (buildIde && !versionAtLeast "8.10")
+  coqAtLeast = v: coq-version == "dev" || versionAtLeast coq-version v;
+  ideFlags = optionalString (buildIde && !coqAtLeast "8.10")
```

Comment:
For these I'm wondering why there's not both `coqAtLeast` and `coqOlder`, now there's still negated usage.

Reply:
Adding `coqOlder` seems like a reasonable further cleanup. But there are other cleanups I’d want to do there too (like…what is up with these bizarre non-transitive [truncating version comparators](https://github.com/NixOS/nixpkgs/blob/87d34a6b8982e901b8e50096b8e79ebc0e66cda0/pkgs/build-support/coq/extra-lib.nix#L4-L68) that get [monkey-patched into `lib`](https://github.com/NixOS/nixpkgs/blob/87d34a6b8982e901b8e50096b8e79ebc0e66cda0/pkgs/applications/science/logic/coq/default.nix#L17)…this is what half-open ranges are for…?). So I’m inclined to defer that discussion to a future Coq-specific PR so as not to bother the other maintainers that got automatically pinged here.

Reply:
Makes sense.